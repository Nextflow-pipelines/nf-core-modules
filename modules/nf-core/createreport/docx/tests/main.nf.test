// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test createreport/docx
nextflow_process {

    name "Test Process CREATEREPORT_DOCX"
    script "../main.nf"
    process "CREATEREPORT_DOCX"

    tag "modules"
    tag "modules_nfcore"
    tag "createreport"
    tag "createreport/docx"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("docx - fastqc") {

        // TODO nf-core: If you are created a test for a chained module
        // (the module requires running more than one process to generate the required output)
        // add the 'setup' method here.
        // You can find more information about how to use a 'setup' method in the docs (https://nf-co.re/docs/contributing/modules#steps-for-creating-nf-test-for-chained-modules).

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:

                input[0] = [
                    file(params.test_data['generic']['yaml']['test_customer_info'], checkIfExists: true)
                    ]
                input[1] = [
                    file(params.test_data['generic']['yaml']['test_software_versions'], checkIfExists: true)
                    ]
                input[2] = [
                    file(params.test_data['generic']['docx']['test_docx'], checkIfExists: true)
                    ]
                input[3] = [
                    file(params.test_data['generic']['yaml']['test_report_contents'], checkIfExists: true)
                    ]
                input[4] = [
                    file(params.test_data['generic']['png']['test_workflow_png'], checkIfExists: true),
                    file(params.test_data['generic']['png']['test_contents_png'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                //{ assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
                //{ assert process.out.report ==~ "*.docx" },
                //{ assert process.out.informations ==~ "all_informations.yml" },
                { assert snapshot(process.out.report).match("*.docx") },
                { assert snapshot(process.out.informations).match("all_informations.yml") },
                { assert snapshot(process.out.versions).match("versions_docx_fastqc") }
            )
        }

    }

}
