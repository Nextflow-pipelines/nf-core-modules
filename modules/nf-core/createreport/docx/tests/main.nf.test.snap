{
    "versions_docx_fastqc": {
        "content": [
            [
                "versions.yml:md5,07a1221c16a713fc5a5eed4680716fc9"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-02T12:03:30.719695"
    },
    "*.docx": {
        "content": [
            [
                "K562\u30fb\u793e\u5185\u8a66\u9a13_CAGE_report_20240302.docx:md5,22f65212297a278bdbfa0df8b155a9d2"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-02T12:06:14.411363"
    },
    "docx - fastqc": {
        "content": [
            {
                "0": [
                    "K562\u30fb\u793e\u5185\u8a66\u9a13_CAGE_report_20240302.docx:md5,300ce1b4540a3775155ceb8f4a9b23ff"
                ],
                "1": [
                    "all_informations.yml:md5,d3b8a23d1f3509717fd8084b9d2a433f"
                ],
                "2": [
                    "versions.yml:md5,07a1221c16a713fc5a5eed4680716fc9"
                ],
                "informations": [
                    "all_informations.yml:md5,d3b8a23d1f3509717fd8084b9d2a433f"
                ],
                "report": [
                    "K562\u30fb\u793e\u5185\u8a66\u9a13_CAGE_report_20240302.docx:md5,300ce1b4540a3775155ceb8f4a9b23ff"
                ],
                "versions": [
                    "versions.yml:md5,07a1221c16a713fc5a5eed4680716fc9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-02T12:04:20.944121"
    },
    "all_informations.yml": {
        "content": [
            [
                "all_informations.yml:md5,d3b8a23d1f3509717fd8084b9d2a433f"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-02T12:03:30.710061"
    }
}